app/services/media_storage/azure_adapter.rb

```diff
@@ -0,0 +1,92 @@
+module MediaStorage
+  class AzureAdapter < AbstractAdapter
+    DEFAULT_EXPIRES_IN = 3 # time in minutes
+
+    def initialize(opts={})
+      @storage_account_name = opts[:azure_storage_account]
+      @container = opts[:azure_storage_container] || Rails.env
+      @get_expiration = opts.dig(:expiration, :get) || DEFAULT_EXPIRES_IN
+      @put_expiration = opts.dig(:expiration, :put) || DEFAULT_EXPIRES_IN
+
+      @client = Azure::Storage::Blob::BlobService.create(
+        storage_account_name: opts[:azure_storage_account],
+        storage_access_key: opts[:azure_storage_access_key]
+      )
+      @signer = Azure::Storage::Common::Core::Auth::SharedAccessSignature.new(
+        opts[:azure_storage_account],
+        opts[:azure_storage_access_key]
+      )
+    end
+
+    def stored_path(content_type, medium_type, *path_prefix)
+      extension = get_extension(content_type)
+      path = "#{medium_type}/"
+      path += "#{path_prefix.join('/')}/" unless path_prefix.empty?
+      path + "#{SecureRandom.uuid}.#{extension}"
+    end
+
+    def get_path(path, opts={})
+      # TO DO: implement private v public uploads
+      expires_in = opts[:get_expires] || @get_expiration # time in minutes
+      expiry_time = get_expiry_time(expires_in)
+
+      @signer.signed_uri(
+        generate_uri(path), false,
```

Comment:
ask the client for the storage URI via client `generate_uri` method vs our own possibly brittle version custom one which is brittle..
https://github.com/Azure/azure-storage-ruby/blob/e5059568b6b971f6981b454fd71cf211429d418e/common/lib/azure/core/service.rb#L39

https://github.com/rails/rails/blob/c9abc6801e0c1dd187f0f7e3ca127a29a6eba1bb/activestorage/lib/active_storage/service/azure_storage_service.rb#L128

Also please use move the `use_account_sas` boolean param to its own line.

Reply:
Sure - just as a side note, the reason I put it on its own line was to mirror how rails used the method; I figured it was a convention around passing unnamed parameters vs named parameters.
https://github.com/rails/rails/blob/badba9d2fa90ae96a5d8428b0a739e18b1581607/activestorage/lib/active_storage/service/azure_storage_service.rb#L113